code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/InputBoxNotificationBuilderTest.kt

```diff
@@ -0,0 +1,180 @@
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class InputBoxNotificationBuilderTest {
+    @Mock
+    private lateinit var context: Context
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        context = RuntimeEnvironment.getApplication()
+        trackerActivityClass = DummyActivity::class.java
+        broadcastReceiverClass = DummyBroadcastReceiver::class.java
+    }
+
+    @Test
+    fun `construct should return a NotificationCompat Builder`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        TestCase.assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test
+    fun `construct should not have any inputText action if the template is created from intent`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData(true)
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertEquals(0, notificationBuilder.mActions.size)
+    }
+
+    @Test
+    fun `construct should have an InputText action if the template is not created from intent`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertEquals(1, notificationBuilder.mActions.size)
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test
+    fun `createInputReceivedPendingIntent should not set class when trackerActivityClass parameter is null`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            null,
+            broadcastReceiverClass
+        )
+
+        val notification = notificationBuilder.build()
+        val pendingIntent = notification.contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertNotNull(intent)
+        assertNull(intent.resolveActivity(context.packageManager))
+    }
+
+    @Test
+    fun `createInputReceivedPendingIntent should set class when trackerActivityClass parameter is null`() {
```

Comment:
nit: `createInputReceivedPendingIntent should set class when trackerActivityClass parameter is **not** null`